### PR TITLE
feat: add db triggers to handle upvotes count

### PR DESCRIPTION
**Pull request by capJavert**
As per comments in the [DR](https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/544768013/Downvoting+and+reporting+posts#Other-notes) replacing inline increments with db triggers to avoid race conditions while upvoting and in the future downvoting.

Also added downvote triggers.

Also updated two tests that were right but expected wrong values due to test setup, see 4962c92f3eac11f3af9ab6e4ff06fe9f3ec5be33 commit message.

As additional note, in general we could apply this for comments too.

### CODE REVIEW

File: src/migration/1686749310179-UpvoteTrigger.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        BEGIN
+          UPDATE post SET upvotes = upvotes + 1 WHERE id = NEW."postId";
+          RETURN NEW;
+        END;
+        $$
+      `)
+      queryRunner.query('CREATE TRIGGER upvote_create_trigger AFTER INSERT ON "upvote" FOR EACH ROW EXECUTE PROCEDURE increment_upvote()')
```
Comment (idoshamun) [inline src/migration/1686749310179-UpvoteTrigger.ts:19]: CREATE OR REPLACE may be better just in case. Also for the other trigger

Comment (capJavert) [reply src/migration/1686749310179-UpvoteTrigger.ts:19]: @idoshamun yeah I tried it but `OR REPLACE` for `CREATE TRIGGER` is only available from version [14](https://www.postgresql.org/docs/14/sql-createtrigger.html), we are on version [11](https://www.postgresql.org/docs/11/sql-createtrigger.html) from what I can see.

Comment (idoshamun) [reply src/migration/1686749310179-UpvoteTrigger.ts:19]: Oh right! Thanks for letting me know

**Review by idoshamun — APPROVED** [review]

**Review by rebelchris — APPROVED** [review]
Nice, yeah these functions are pretty cool actually.

**Pull request merged**